compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUtils.cpp

```diff
@@ -298,33 +287,46 @@ bool isMatmul(linalg::LinalgOp linalgOp) {
          match6DLinalgGenericMatmul(linalgOp);
 }
 
+/// Utility to identify if the input operand has matmul-like op in its
+/// def-chain.
+bool isMatmulInDefChain(Value operand) {
+  while (Operation *defOp = operand.getDefiningOp()) {
+    if (isa<arith::ConstantOp>(defOp)) return false;
+    if (auto defLinalgOp = dyn_cast_or_null<linalg::LinalgOp>(defOp)) {
+      if (isMatmul(defLinalgOp)) {
+        return true;
+      }
+      break;
+    }
+    if (auto forOp = dyn_cast_or_null<scf::ForOp>(defOp)) {
+      bool findMatmul = false;
+      forOp.getBody()->walk<WalkOrder::PostOrder, ReverseIterator>(
+          [&](linalg::LinalgOp op) {
+            if (isMatmul(op)) {
+              findMatmul = true;
+              return WalkResult::interrupt();
+            }
+            return WalkResult::advance();
+          });
+      if (findMatmul) {
+        return true;
+      }
+    }
+    operand = defOp->getOperand(0);
```

Comment:
Why do we only traverse the def-chain of the first operand here?

I understand that any LinalgOp which is not matmul-like will result in `false`, but what other operations besides `scf::ForOp` can we get in the def-chain whose first operand holds the relevance? I can only think of `tensor.pack/unpack`.

Reply:
Normally the defining op should be tensor.pack or tensor.extract_slice, and we only take the source operand from it.

Reply:
I have the same question. Maybe some more context here would help. Ideally we dont want such a walk here. Its seems a bit fragile.